draft-ietf-dnsop-structured-dns-error-page.xml

```diff
@@ -0,0 +1,1003 @@
+      pointing at a host operated by the DNS operator.  The
+      expectation is the user's browser will visit the server over
+      HTTP and a suitable explanation will be displayed to the user.
+      As HTTPS is now commonplace, this expectation breaks down because
+      clients are presented with a warning about visiting an
+      un-trusted site (because of an un-trusted certificate), or those
+      users are encouraged to install the certificate as an additional
+      trusted root, eroding their security for all TLS connections.
+      Furthermore, for the user to see the DNS filtering explanation
+      the user needs a web browser which visits the server and often
+      require JavaScript, creating a wider attack surface.  The
+      requirement for a web browser is, itself, a problem on some
+      devices (e.g., IoT or headless devices).  Finally, this mechanism
+      interferes with client DNSSEC-validating DNS resolvers because
+      the forged response has an invalid (or missing) signature.</dd>
+      </dl>
+      -->
+
+      <t />
+
+      <ol>
+        <li>
+          <!--  Neil suggested removing these couple of sentences
+
+Most importantly, EDE option does not offer
+   authenticated information; it can thus be spoofed by an attacker.
+   In addition, the additional textual information may not be able to
+   convey all of the required information about the cause of the DNS
+   error because lengthy EXTRA-TEXT content would be truncated to
+   prevent fragmentation (<xref target="RFC8914">Section 3 of </xref>).
+-->
+        </li>
+      </ol>
+
+      <t>For both DNS filtering mechanisms described in <xref
+      target="filtering">Section 4 of</xref>, the DNS server can return
+      extended error codes Blocked, Censored, Filtered, or Forged Answer
+      defined in <xref target="RFC8914">Section 4 of</xref>. However, these
+      codes only explain that filtering occurred but lack detail for the user
+      to diagnose erroneous filtering.</t>
+
+      <t>No matter which type of response is generated (forged IP address(es),
+      NXDOMAIN or empty answer, even with an extended error code), the user
+      who triggered the DNS query has little chance to understand which entity
+      filtered the query, how to report a mistake in the filter, or why the
+      entity filtered it at all. This document describes a mechanism to
+      provide such detail.</t>
+
+      <t>One of the other benefits of this approach is to eliminate the need
+      to "spoof" block pages for HTTPS resources. This is achieved since
+      clients implementing this approach would be able to display a meaningful
+      error message, and would not need to connect to such a block page. This
+      approach thus avoids the need to install a local root certificate
+      authority on those IT-managed devices.</t>
+
+      <t>This document describes a format for computer-parsable data in the
```

Comment:
We need to add a warning here that this part is updating 8914, which says: 

"This information is intended for human consumption (not automated parsing)."